clab/clab.go

```diff
@@ -457,30 +458,32 @@ func (c *CLab) scheduleNodes(ctx context.Context, maxWorkers int,
 	// Waitgroup used to protect the channel towards the workers of being closed to early
 	workerFuncChWG := new(sync.WaitGroup)
 
-	for _, n := range scheduledNodes {
-		workerFuncChWG.Add(1)
-		// start a func for all the containers, then will wait for their own waitgroups
-		// to be set to zero by their depending containers, then enqueue to the creation channel
-		go func(node nodes.Node, dm DependencyManager, workerChan chan<- nodes.Node, wfcwg *sync.WaitGroup) {
-			// wait for all the nodes that node depends on
-			err := dm.WaitForNodeDependencies(node.Config().ShortName)
-			if err != nil {
-				log.Error(err)
-			}
-			// wait for possible external dependencies
-			c.WaitForExternalNodeDependencies(ctx, node.Config().ShortName)
-			// when all nodes that this node depends on are created, push it into the channel
-			workerChan <- node
-			// indicate we are done, such that only when all of these functions are done, the workerChan is being closed
-			wfcwg.Done()
-		}(n, dm, concurrentChan, workerFuncChWG) // execute this function straight away
-	}
-
-	// Gate to make sure the channel is not closed before all the nodes made it though the channel
-	workerFuncChWG.Wait()
-	// close the channel and thereby terminate the workerFuncs
-	close(concurrentChan)
+	// schedule nodes via a go func to create links in parallel
+	go func() {
+		for _, n := range scheduledNodes {
```

Comment:
@steiler what is the purpose of this goroutine? Does it make things faster in any way, because I see that we start a goroutine on ln467 anyhow, so I don't quite the optimization here

Reply:
In 483 we wait for the nodes to make it through the channel before closing it.
That means, that the Link deployment cannot start, since the Call from Schedule nodes has to return.
See
https://github.com/srl-labs/containerlab/blob/4ea14a31b24d978ae3c005f2f299e4a9c5c9294e/cmd/deploy.go#L223-L230